package.json

```diff
@@ -108,13 +120,19 @@
     "sinon": "^1.17.4",
     "storybook-addon-react-docgen": "^1.0.4",
     "style-loader": "^0.23.1",
+    "tsd": "^0.11.0",
+    "typescript": "^3.9.5",
     "uglifyjs-webpack-plugin": "^2.0.1",
     "url-loader": "^1.1.2",
     "webpack": "^4.27.1"
   },
   "scripts": {
     "prebuild": "yarn run clean",
-    "build": "BABEL_ENV=production babel src -d dist && cp -r ./src/*.png ./dist && node-sass ./src/intlTelInput.scss ./dist/main.css",
+    "build": "yarn build:js && yarn build:png && yarn build:css && yarn build:ts",
+    "build:js": "cross-env BABEL_ENV=production babel src -d dist",
+    "build:png": "cp -r ./src/*.png ./dist",
+    "build:css": "node-sass ./src/intlTelInput.scss ./dist/main.css",
+    "build:ts": "gulp build:ts",
```

Comment:
I've split this up because it was too long to read in my screen, especially now with the `gulp` build added.